app/src/main/java/org/mozilla/fenix/compose/tabstray/ListTabItem.kt

```diff
@@ -0,0 +1,170 @@
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Card
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.tabstray.TabsTray
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.TabImage
+import org.mozilla.fenix.compose.TabTitle
+import org.mozilla.fenix.compose.TabUrl
+import org.mozilla.fenix.ext.components
+
+@Composable
+fun ListTabItem(
+    tab: TabSessionState,
+    isSelected: Boolean = false,
+    multiSelectionEnabled: Boolean = false,
+    multiSelectionSelected: Boolean = false,
+    delegate: TabsTray.Delegate? = null
+) {
+    val contentBackgroundColor = if (isSelected) {
+        R.color.tab_tray_item_selected_background_normal_theme
+    } else {
+        R.color.tab_tray_item_background_normal_theme
+    }
+    val descriptionWeight = if (multiSelectionEnabled) 4f else 3f
+
+    Row(modifier = Modifier
+        .fillMaxWidth()
+        .height(dimensionResource(id = R.dimen.tab_tray_tab_item_height))
+        .background(colorResource(id = contentBackgroundColor))) {
+
+        Thumbnail(tab, multiSelectionSelected)
+
+        DescriptionSection(descriptionWeight, tab)
+
+        CloseIcon(
+            tab = tab,
+            visible = !multiSelectionEnabled,
+            delegate = delegate
+        )
+    }
+}
+
+@Composable
+private fun RowScope.Thumbnail(tab: TabSessionState, isSelected:Boolean) {
+    Box(
+        modifier = Modifier.Companion
+            .weight(2f)
+            .padding(start = 16.dp, end = 12.dp, top = 8.dp, bottom = 8.dp)
+    ) {
+        TabImage(
+            tab = tab,
+            modifier = Modifier
+                .size(
+                    dimensionResource(id = R.dimen.tab_tray_list_item_thumbnail_width),
```

Comment:
I can in-vision a world where we simply keep the dimension values inside of the Composable function. In factor, we do a mix of this in our layout xml and Composable function currently. Ideally, when we drop the layout xml for the tab list item, let's also remove the value defined in dimens.xml if we only need the value within the Composable function.